packages/firestore/src/local/simple_db.ts

```diff
@@ -70,7 +76,17 @@ export class SimpleDb {
         // cheating and just passing the raw IndexedDB in, since
         // createObjectStore(), etc. are synchronous.
         const db = (event.target as IDBOpenDBRequest).result;
-        runUpgrade(db, event.oldVersion);
+        runUpgrade(
+          db,
+          request.transaction,
```

Comment:
As mentioned elsewhere, I think we should wrap this in a PersistenceTransaction.  

And please remove / update the above TODO as appropriate.

Reply:
Done.